network/src/connectivity_manager/mod.rs

```diff
@@ -338,6 +362,30 @@ where
             })
             .collect();
 
+        // Limit the number of dialed connections from a Full Node
+        // This does not limit the number of incoming connections
+        // It enforces that a full node cannot have more outgoing connections than `connection_limit`
+        // including in flight dials.
+        let to_connect_size = if let Some(conn_limit) = self.connection_limit {
+            min(
+                conn_limit - self.connected.len() - self.dial_queue.len(),
+                to_connect.len(),
+            )
+        } else {
+            to_connect.len()
+        };
+
+        // If there's nothing to dial, let's just skip the rest of the function
+        if to_connect_size == 0 {
+            return;
+        }
+        to_connect.choose_multiple(&mut self.rng, to_connect_size);
+        let to_connect = &to_connect[0..to_connect_size];
+
+        send_struct_log!(StructuredLogEntry::new_named("dial_eligible_peers")
+            .field(&logging::NETWORK_CONTEXT, &self.network_context)
+            .data("new_peers", &to_connect));
+
```

Comment:
promise me you'll send a pr to delete this blob below <3

Reply:
I will send a separate PR for it!

Reply:
Here it is!https://github.com/libra/libra/pull/4684